cumulusci/tests/pytest_plugins/pytest_sf_orgconnect.py

```diff
@@ -1,27 +1,76 @@
-import os.path
+from contextlib import contextmanager
+from contextvars import ContextVar
+from pathlib import Path
+from unittest import mock
 
+import click
 import pytest
 
+import cumulusci
+from cumulusci.cli.org import org_scratch, org_scratch_delete
 from cumulusci.cli.runtime import CliRuntime
 from cumulusci.core.config import TaskConfig
+from cumulusci.core.exceptions import OrgNotFound
 from cumulusci.salesforce_api.utils import get_simple_salesforce_connection
 from cumulusci.tests.util import unmock_env
 
+from .pytest_sf_vcr import READING, RECORDING, set_mode
+
 
 def pytest_addoption(parser, pluginmanager):
+    """Pytest magic method: add --org and --run-slow-tests features"""
     parser.addoption("--org", action="store", default=None, help="org to use")
     parser.addoption(
-        "--accelerate-integration-tests",
+        "--run-slow-tests",
+        action="store_true",
+        default=False,
+        help="Include slow tests.",
+    )
+    parser.addoption(
+        "--replace-vcrs",
         action="store_true",
         default=False,
-        help="DO run integration tests. Do NOT make calls to a real org. This will error out if you have not run with '--org blah' so that you have cached org output.",
+        help="Replace VCR files.",
     )
 
 
-def sf_pytest_orgname(request):
+def pytest_configure(config):
+    """Pytest magic method: add pytest markers/decorators"""
+    markers = [
+        "slow(): a slow test that should only be executed when requested with --run-slow-tests",
+        "large_vcr(): a network-based test that generates VCR cassettes too large for version control. Use --org to generate them locally.",
+        "needs_org(): a test that needs an org (or at least access to the network) but should not attempt to store VCR cassettes",
+        "org_shape(org_name, init_flow): a test that needs a particular org shape",
+    ]
+
+    for marker in markers:
+        # register additional markers
+        config.addinivalue_line("markers", marker)
+
+
+def pytest_runtest_setup(item):
+    """Pytest magic method"""
+    marker_names = set(marker.name for marker in item.iter_markers())
+    classify_and_modify_test(item, marker_names)
+
+
+def sf_pytest_cli_orgname(request):
+    """Helper to see what org the user has ased for"""
     return request.config.getoption("--org")
 
 
+@pytest.fixture(scope="session")
+def user_requested_network_access(request):
```

Comment:
The "network access" terminology I find confusing. There doesn't appear to be any distinction between network access and access to an org. Is that accurate?

Reply:
Seems weird to need to specify an org in order to run an integration test for a task that only talks to Marketing Cloud or github, but ok